### DISCUSSION TITLE
Changing the mailing server

### DISCUSSION THREAD

**Discussion by sametates4** [saran-surya/email_auth_node#29] [Q&A]
hello @saran-surya  , I want to make the mail sending server on my own server instead of heroku, how can I do this.

**Comment by saran-surya** [comment]
Hi @sametates4 , the CLI will help you to set up the key and the config file, you can directly deploy the node js app on your own server. That should be working fine.

**Reply by sametates4** [reply]
After configuring the files I uploaded "https://api.sametates.com/auth-main/server/index.js" here on my server. I get an error when I put this link as the server url. What exactly should I put?

**Reply by saran-surya** [reply]
Yes it is because you are accessing the file on the server. You will have to run the node js instance and have the parent url only. In that case it will be https://api.sametates.com. but make sure you start the node js instance

**Reply by sametates4** [reply]
I moved the files to a parent node but I still get the error.

**Reply by saran-surya** [reply]
No have you initiated the node instance ?

**Reply by sametates4** [reply]
how do i initialize node instance

**Reply by saran-surya** [reply]
You have to run the the following command in the CLI of the server.
```npm start```

**Comment by sametates4** [comment]
I couldn't do it on my server so I moved it to heroku

**Reply by saran-surya** [reply]
Does your server have node js and python installed ?

**Comment by sametates4** [comment]
When I asked the hosting company, they did not respond, I tried to set up, but I could not succeed.

**Reply by saran-surya** [reply]
do they have a accessible terminal that you can connect through the cloud ?. Also is it cloud hosting or frontend ?

**Comment by sametates4** [comment]
no cloud accessible terminal is available

**Reply by saran-surya** [reply]
Then please check If they are a frontend provider or backend service provider, because you need a backend to run the server.

**Comment by sametates4** [comment]
server frontend provider

**Reply by saran-surya** [reply]
What, can you be a bit more clear, else please give me the provider name, the link to their website.

**Comment by sametates4** [comment]
provider: güzelhosting
url: https://www.guzel.net.tr

There is no area where I can install npm and python in the cPanel interface, it does not exist in a terminal that I can access.

**Reply by saran-surya** [reply]
That's a hosting platform for frontend, you cannot deploy the server there, you will have to choose some cloud backend provider.

**Comment by sametates4** [comment]
OK. thank you for your help. Do you have a recommended server?

**Reply by saran-surya** [reply]
Heroku is free, you can also use AWS or Google cloud
